housing_cooperative_base/models/lease.py

```diff
@@ -337,3 +378,30 @@ def _default_journal(self):
         )
         domain = [("type", "=", "sale"), ("company_id", "=", company_id)]
         return self.env["account.journal"].search(domain, limit=1)
+
+    @api.multi
+    def _automatic_renewal(self):
+        for lease in self:
+            if (
+                lease.state == "ongoing"
+                and lease.automatic_renewal
+                and fields.Date.today() + relativedelta(days=1)
+                >= lease.expected_end
+            ):
+                # If the end date was the last day of the month, keep it as such
+                if lease.expected_end == lease.expected_end + relativedelta(
+                    day=31
+                ):
```

Comment:
I don't understand how this expression could ever be true :/

Reply:
since it says `day` and not `days`, `relativedelta` moves it to the last day of the months => This is `True` if `lease.expected_end` is the last day of the month.

Reply:
Ah, interesting, I didn't know the difference between the day and days arg 👍 

(actually, I never use relativedelta, I should look into it)